src/nodes/CodeBlockView.vue

```diff
@@ -0,0 +1,214 @@
+						<template #icon>
+							<Eye :size="20" />
+						</template>
+						{{ t('text', 'Diagram') }}
+					</NcActionButton>
+					<NcActionButton v-if="supportPreview" :close-after-click="true" @click="viewMode = 'side-by-side'">
+						<template #icon>
+							<ViewSplitVertical :size="20" />
+						</template>
+						{{ t('text', 'Both') }}
+					</NcActionButton>
+
+					<NcActionSeparator v-if="supportPreview" />
+
+					<NcActionLink v-if="supportPreview" href="https://mermaid.js.org/intro/" target="_blank">
+						<template #icon>
+							<Help :size="20" />
+						</template>
+						{{ t('text', 'Mermaid documentation') }}
+					</NcActionLink>
+				</NcActions>
+			</div>
+		</div>
+		<div :class="{'split-view': showCode && showPreview }">
+			<pre v-show="showCode" class="split-view__code"><NodeViewContent as="code" :contenteditable="editable" /></pre>
+			<div v-show="showPreview"
+				ref="preview"
+				class="split-view__preview"
+				:contenteditable="false" />
+			<div v-show="false" :id="targetId" :contenteditable="false" />
+		</div>
+	</NodeViewWrapper>
+</template>
+
+<script>
+import debounce from 'debounce'
+import { NodeViewWrapper, NodeViewContent } from '@tiptap/vue-2'
+import { NcActions, NcActionButton, NcActionInput, NcActionLink, NcActionSeparator } from '@nextcloud/vue'
+import mermaid from 'mermaid'
+import { v4 as uuidv4 } from 'uuid'
+
+import ViewSplitVertical from 'vue-material-design-icons/ViewSplitVertical.vue'
+import CodeBraces from 'vue-material-design-icons/CodeBraces.vue'
+import Eye from 'vue-material-design-icons/Eye.vue'
+import MarkerIcon from 'vue-material-design-icons/Marker.vue'
+import Help from 'vue-material-design-icons/Help.vue'
+
+export default {
+	// eslint-disable-next-line vue/match-component-file-name
+	name: 'CodeBlockView',
+	components: {
+		MarkerIcon,
+		Help,
+		Eye,
+		ViewSplitVertical,
+		CodeBraces,
+		NcActions,
+		NcActionButton,
+		NcActionInput,
+		NcActionLink,
+		NcActionSeparator,
+		NodeViewWrapper,
+		NodeViewContent,
+	},
+	props: {
+		node: {
+			type: Object,
+			required: true,
+		},
+		editor: {
+			type: Object,
+			required: true,
+		},
+	},
+	data() {
+		return {
+			isEditable: false,
+			viewMode: 'preview',
+			targetId: 'mermaid-view-' + uuidv4(),
+		}
+	},
+	computed: {
+		type() {
+			return this.node?.attrs?.language || ''
+		},
+		supportPreview() {
+			return ['mermaid'].includes(this.type)
+		},
+		showCode() {
+			return !this.supportPreview || this.viewMode === 'code' || this.viewMode === 'side-by-side'
+		},
+		showPreview() {
+			return this.supportPreview && (this.viewMode === 'preview' || this.viewMode === 'side-by-side')
+		},
+	},
+	watch: {
+		'node.textContent'() {
+			this.renderMermaid()
+		},
+	},
+	beforeMount() {
+		this.isEditable = this.editor.isEditable
+		this.editor.on('update', ({ editor }) => {
+			this.isEditable = editor.isEditable
+		})
+		this.renderMermaidDebounced = debounce(async function() {
```

Comment:
It should work to define `renderMermaid: debounce(async function() { ... })` in `methods`.

Reply:
I didn't manage to do that as the this context was always missing then

Reply:
I see, then let's keep it as is for now :relaxed: